github/commitstatus.go

```diff
@@ -111,18 +113,23 @@ func (s CommitStatus) Add(sha, state, targetURL, description string) error {
 	var response httpResponse
 	timeToSleep := 0 * time.Second // 0 seconds
 
+retryLoop:
 	for attempt := 1; attempt <= s.target.MaxRetries; attempt++ {
 		time.Sleep(timeToSleep)
 		s.log.Info("GitHub HTTP request", "attempt", attempt, "max", s.target.MaxRetries)
 		response, err = httpRequestDo(req)
 		if err != nil {
 			return err
 		}
-		timeToSleep = checkForRetry(response, s.target.WaitTime, s.target.MaxSleepTime)
-		if timeToSleep == 0 {
-			break
+		timeToSleep = checkForRetry(response, s.target.WaitTime, s.target.MaxSleepTime, s.target.Jitter)
+		switch {
+		case timeToSleep == 0:
+			break retryLoop
+		case timeToSleep < 0:
+			s.log.Warn("Sleep time should not have a negative value", "time", timeToSleep)
```

Comment:
I was also thinking if we should just return an error here

Reply:
```go
s.log.Warn("unexpected: negative sleep time", "time", timeToSleep)
```